src/utils.jl

```diff
@@ -174,6 +174,8 @@ function getsource(source, use_mmap)
         return source
     elseif source isa Cmd
         return Base.read(source)
+    elseif source isa AbstractPath
+        return Base.read(Base.open(source))
```

Comment:
Hmmm, I don't love this because we're not matching the behavior when `source` is a String, which I think should be consistent. When `source` is a String and we have `use_mmap`, then we `Mmap.mmap`, otherwise, we make our own mmapped vector and copy the file contents into it. I'd rather rearrange to ensure strings and abstractpath sources are treated the same.

Reply:
Note that some FilePaths can be `mmap`s directly, in particular `SystemPath`s
other cannot, like `AWS3.S3Path`s
But I guess that can be fairly easily handled by treating `SystemPath`s like strings,
and all other `AbstractPaths` like `Cmd`s etc.

https://github.com/rofinn/FilePathsBase.jl/issues/51

Reply:
`SystemPath`s don't directly support `mmap` and have to be opened first. We're also stuck using open on non-System Paths since that guarantees us the right type coming out of `read` (`S3Path` right now returns a `String`). 

Also why are we checking `slurp(source isa IO ? source : open(String(source)))` after the `elseif !isa(source, IO)` in this function?